src/app/modules/shared/components/feedback/feedback.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, ValidationErrors, Validators } from '@angular/forms';
+import { Observable, Observer } from 'rxjs';
+
+@Component({
+  selector: 'app-feedback',
+  templateUrl: './feedback.component.html',
+  styleUrls: ['./feedback.component.scss']
+})
+export class FeedbackComponent {
+  validateForm!: FormGroup;
+  isVisible = false;
+  isOkLoading = false;
+
+  constructor(private fb: FormBuilder) {
+    this.validateForm = this.fb.group({
+      userName: ['', [Validators.required], [this.userNameAsyncValidator]],
+      email: ['', [Validators.email, Validators.required]],
+      password: ['', [Validators.required]],
+      confirm: ['', [this.confirmValidator]],
+      comment: ['', [Validators.required]]
+    });
+  }
+
+  showModal(): void {
+    this.isVisible = true;
+  }
```

Comment:
I guess that Input() decorator is much more approriate (there is no complex logic in child component that should be implemented during parent's working process).

Reply:
I didn't catch this point)